weblab_cg/data/tests/chaste_reference_models/Normal/demir_model_1994.hpp

```diff
@@ -1,78 +1,78 @@
-#include "AbstractCardiacCell.hpp"
-#include "AbstractDynamicallyLoadableEntity.hpp"
-#include "AbstractStimulusFunction.hpp"
-
-class Dynamicdemir_model_1994FromCellML : public AbstractCardiacCell, public AbstractDynamicallyLoadableEntity
-{
-    friend class boost::serialization::access;
-    template<class Archive>
-    void serialize(Archive & archive, const unsigned int version)
-    {
-        archive & boost::serialization::base_object<AbstractCardiacCell >(*this);
-        archive & boost::serialization::base_object<AbstractDynamicallyLoadableEntity>(*this);
-    }
-    
-    // 
-    // Settable parameters and readable variables
-    // 
-    
-public:
-    double GetIntracellularCalciumConcentration();
-    Dynamicdemir_model_1994FromCellML(boost::shared_ptr<AbstractIvpOdeSolver> pSolver, boost::shared_ptr<AbstractStimulusFunction> pIntracellularStimulus);
-    ~Dynamicdemir_model_1994FromCellML();
-    double GetIIonic(const std::vector<double>* pStateVariables=NULL);
-    void EvaluateYDerivatives(double var_chaste_interface__environment__time, const std::vector<double>& rY, std::vector<double>& rDY);
-};
-
-
-// Needs to be included last
-#include "SerializationExportWrapper.hpp"
-CHASTE_CLASS_EXPORT(Dynamicdemir_model_1994FromCellML)
-
-namespace boost
-{
-    namespace serialization
-    {
-        template<class Archive>
-        inline void save_construct_data(
-            Archive & ar, const Dynamicdemir_model_1994FromCellML * t, const unsigned int fileVersion)
-        {
-            const boost::shared_ptr<AbstractIvpOdeSolver> p_solver = t->GetSolver();
-            const boost::shared_ptr<AbstractStimulusFunction> p_stimulus = t->GetStimulusFunction();
-            ar << p_solver;
-            ar << p_stimulus;
-        }
-        
-        template<class Archive>
-        inline void load_construct_data(
-            Archive & ar, Dynamicdemir_model_1994FromCellML * t, const unsigned int fileVersion)
-        {
-            boost::shared_ptr<AbstractIvpOdeSolver> p_solver;
-            boost::shared_ptr<AbstractStimulusFunction> p_stimulus;
-            ar >> p_solver;
-            ar >> p_stimulus;
-            ::new(t)Dynamicdemir_model_1994FromCellML(p_solver, p_stimulus);
-        }
-        
-    }
-    
-}
-
-#endif // DYNAMICDEMIR_MODEL_1994FROMCELLML_HPP_
+#ifndef TESTMANUALDEMIR_MODEL_1994FROMCELLML_HPP_
+#define TESTMANUALDEMIR_MODEL_1994FROMCELLML_HPP_
+
+//! @file
+//! 
+//! This source file was generated from CellML.
+//! 
+//! Model: demir_model_1994
+//! 
+//! Processed by weblab_cg: https://github.com/ModellingWebLab/weblab-cg
+//!     (translator: weblab_cg, model type: normal, options: )
+//! on 2019-11-27 16:18:14
+//! 
+//! <autogenerated>
+
+#include "ChasteSerialization.hpp"
+#include <boost/serialization/base_object.hpp>
+#include "AbstractCardiacCell.hpp"
+
+#include "AbstractStimulusFunction.hpp"
+
+class TestManualdemir_model_1994FromCellML : public AbstractCardiacCell

```

Comment:
I'm not sure about the `Test` prefix on model class names?

Reply:
It's just a name picked by the test suite! We can change it of course let me know if you think it's important. Not for this PR though!

Reply:
Yeah, this is very confusing for Chaste code as things starting with Test should actually be tests!

Reply:
what do you both suggest as prefix? (currently `TestManual`)

Reply:
If will want to be the same as PyCml currently does won't it - `Cell` ? Our is this an intermediate step so we can generate both? If so `CellmlmanipCell` ??

Reply:
don't confuse these tests with Chaste tests though. This is just a textual comparison on the generated cpp / hpp against a reference, the name is just a variable in the test (we we hook it up the names will be whatever they used to be)

Reply:
Well, if you use the same names as pycml, you don't need to map! Also some (albeit not much) Chaste code might make assumptions about the naming?

Reply:
what do you mean map?

Reply:
Sorry, comment below was sent by email and didn't get attached here for some reason. I understand now. Think I'd keep the `Cell` prefix though.

Reply:
ok!